bin/utils/utils.js

```diff
@@ -1,7 +1,74 @@
+
+const axios = require("axios");
+const chalk = require("chalk");
+
 class FlightTrackerCLI {
-  constructor() {}
+  constructor() {
+    this.axiosInstance = axios.create({
+      baseURL: 'https://api.aviationstack.com/v1',
+      timeout: 5000,
+      headers: {
+        'Content-Type': 'application/json',
+      }
+    });
+  }
+
+  async findIATACode(airportName) {
+    try {
+      const response = await this.axiosInstance.get('/airports', {
+        params: {
+          city: airportName,
+          // access_key: 'YOUR_API_KEY', // option to pass key manually
+        },
+      });
+
+      if (response && response.status === 200 && response.data) {
+        if(response.data.data && response.data.data.length > 0) {
+          console.log(`IATA Code for ${airportName}: ${response.data.data[0].iata_code}`);
+        }
+        else {
+          console.log(chalk.red(`No IATA code found for the given airport name: ${airportName}`));
+        }
+      }
+      else {
+        console.log(chalk.red("Error: Invalid response from API"));
+      }
+
+    } catch (error) {
+      console.error('Error fetching IATA code:', error.message || error);
+    }
+  }
+
+  async findFlights(iataCode) {
+    try {
+      const response = await this.axiosInstance.get('flights', {
+        params: {
+          airport: iataCode,
+        },
+      });
+
+      // return API response data or load it
+      if (response && response.status === 200) {
+        console.log(`Flights at ${iataCode}:`);
+        console.log(response.data); // log the data
+        return response.data;
+      }
+      else {
+        console.log(chalk.red("Error: Failed to retrieve flight data"));
+      }
+    } catch (error) {
+      console.error('Error fetching flights:', error.message || error)
+    }
+  }
 
-  login(manualLogin = false, userName = "", passWord = "") {}
+  async login(manualLogin = false, userName = "", passWord = "") {
+    if (manualLogin) {
+      // manual logic
```

Comment:
Thanks for writing these notes. I'm still toying around on how to handle manually passing the api key. Any ideas would be appreciated here #8